gui/default/syncthing/core/pathIsSubDirDirective.js

```diff
@@ -0,0 +1,38 @@
+angular.module('syncthing.core')
+    .directive('pathIsSubDir', function () {
+        return {
+            require: 'ngModel',
+            priority: 1000,
+            link: function (scope, elm, attrs, ctrl) {
+                ctrl.$parsers.unshift(function (viewValue) {
+                    // This function checks whether xdir is a subdirectory of ydir,
+                    // e.g. it would return true if xdir = "/home/a", ydir = "/home/a/b".
+                    function isSubDir(xdir, ydir) {
+                        var xdirArr = xdir.split("/");
```

Comment:
Doesn't work for Windows folks :(

There is a pathSeparator attribute somewhere in the controller scope.